docs/rfc/rfc-026-p2p-bad-peers-checktx.md

```diff
@@ -0,0 +1,375 @@
+# RFC 26: Banning peers based on ResponseCheckTx
+
+## Changelog
+- Nov 4, 2022: Initial draft (jmalicevic)
+- Nov 8, 2022: Updated draft (jmalicevic)
+
+## Abstract
+
+Before adding a transaction to its mempool, a node runs `CheckTx` on it. `CheckTx` returns a non-zero code
+in case the transaction fails an application specific check. 
+
+Note that there are valid scenarios in which a transaction will not
+pass this check (including nodes getting different transactions at different times, meaning some
+of them might be obsolete at the time of the check; state changes upon block execution etc.). 
+However, Tendermint users observed that
+there are transactions that can never have been or will never be valid. They thus propose
+to introduce a special response code for `CheckTx` to indicate this behaviour, and ban the peers
```

Comment:
If I understand correctly, `CheckTx` is under the control of the application developer. The biggest risk I see is that a bug in `CheckTx` (always failing with an error > 1), will result in everyone banning everyone.

If this is the case, this risk/responsibility of the application developer needs to be properly documented.

Reply:
Another option would be to tie the code to a codespace. There is still the risk of clash, since we didn't reserve codespaces either, but it is much easier to chose a code space that was almost certainly never used before.

Reply:
Why not add a new peer field. 0 is no ban, 1 is decrement reputation, 2 ban right away. This is backwards compatible too since 0 is default if not populated